scipy/optimize/tests/test_optimize.py

```diff
@@ -2612,7 +2612,7 @@ def check_limits(self, method, default_iters):
 
 def test_result_x_shape_when_len_x_is_one():
     def fun(x):
-        return x * x
+        return x[0] * x[0]
```

Comment:
I made the PR to flush these out.  To be fair, optimizers are a bit of a reasonable use-case, since here you probably know that the target to minimize *must* be a scalar so squeezing for the user is maybe OK.

OTOH, if you fix it here, it means the user must do the squeezing, while currently squeezing must be happening by accident somewhere in `optimize.minimize`.

Reply:
I think I agree it would be great to flush the issues out with a traceback to the actual line with the problem in the source. I'm not getting that behavior just yet with that PR per https://github.com/numpy/numpy/pull/24378#pullrequestreview-1570285902, unless I'm misunderstanding.

Reply:
No, I don't think you are.  It still smells like a try/except block causing this, but maybe it is not the one in `np.float64` but another one possibly even in SciPy C code?  I will search a bit, but not sure.

Reply:
Ohhh, these are f2py routines, eh?  I am not sure we should be surprised about a weird try/except in f2py sources :).

Reply:
Yeah, if you look for `double_from_pyobj` that seems like our culprit, it does some weird try/except chaining including `float(sequence[0])` and getting the real part of an imaginary value...

Reply:
I wonder if SciPy should basically just "do nothing" then, or possibly just add small comments about this in the related codepaths for now, if it is basically an `f2py` matter. Or if the argument could be made that we should "squeeze before f2py?"

Reply:
If you *want* reliable squeezing, then maybe...  But overall, even in that case you can argue to just wait for f2py to actually deprecate it and force you to add it.

Reply:
> these are f2py routines?

The minimisers being tested in this file are a mix of pure Python, C, and Fortran.

> you probably know that the target to minimize must be a scalar so squeezing for the user is maybe OK

In the past we've had a lot of issues with users returning a variety of things from their objective functions, and we've had to deal with it. See https://github.com/scipy/scipy/blob/main/scipy/optimize/_differentiable_functions.py#L146 which demonstrates the hoops that have to be jumped through. e.g. we have try to deal with `1.0, np.array(1), np.array([1]), np.float64(1.0)`. An unstated side-effect of this test will be examining that the minimisers work when `np.array([1])` is returned from a function (i.e. size 1, ndim 1). This is quite a frequent user occurrence, even though the documentation states a float is to be returned.
Making the change will remove the (possibly unintended) check - I can't remember if it's checked elsewhere. There's a historical conversation somewhere about using `np.squeeze` in such situations - various people raised objections to it.

The actual aim of the test is to make sure that if you pass in an `x0` array with ndim=1, len=1, then you still get the same type of array returned.

Reply:
It's not